code/game/objects/machinery/vending/quick_vendor.dm

```diff
@@ -202,8 +253,8 @@ GLOBAL_LIST_INIT(quick_loadouts, init_quick_loadouts())
 			if(selected_loadout.jobtype != user_id.rank)
 				to_chat(usr, span_warning("You are not in the right job for this loadout!"))
 				return
-			if(user_id.can_buy_loadout)
-				user_id.can_buy_loadout = FALSE
+			if(user_id.flags_id & CAN_BUY_LOADOUT)
+				user_id.flags_id &= ~CAN_BUY_LOADOUT
```

Comment:
I am 90% sure the way you've coded this means you can open the beginner vendor UI, use the gimmie, then vend a beginner loadout. Maybe vice versa as well, but you'd have to check.